torchtext/csrc/sentencepiece.cpp

```diff
@@ -39,7 +39,59 @@ struct SentencePiece : torch::CustomClassHolder {
     return processor_.EncodeAsPieces(input);
   }
 
+  std::vector<std::vector<std::string>>
+  NBestEncodeAsPieces(const std::string &input,
+                      const int64_t nbest_size) const {
+    return processor_.NBestEncodeAsPieces(input, nbest_size);
+  }
+
+  std::vector<std::vector<int64_t>>
+  NBestEncodeAsIds(const std::string &input, const int64_t nbest_size) const {
+    const auto vals = processor_.NBestEncodeAsIds(input, nbest_size);
+    std::vector<std::vector<std::int64_t>> ret;
+    for (const auto &vec : vals) {
+      ret.emplace_back(std::vector<std::int64_t>(vec.begin(), vec.end()));
+    }
+    return ret;
+  }
+
+  std::vector<std::string> SampleEncodeAsPieces(const std::string &input,
+                                                const int64_t nbest_size,
+                                                const double alpha) const {
+    return processor_.SampleEncodeAsPieces(input, nbest_size, alpha);
+  }
+
+  std::vector<int64_t> SampleEncodeAsIds(const std::string &input,
+                                         const int64_t nbest_size,
+                                         const double alpha) const {
+    const auto val = processor_.SampleEncodeAsIds(input, nbest_size, alpha);
+    return std::vector<int64_t>(val.begin(), val.end());
+  }
+
+  std::string DecodePieces(const std::vector<std::string> &input) const {
+    return processor_.DecodePieces(input);
+  }
+
+  std::string DecodeIds(const std::vector<int64_t> &input) const {
+    const std::vector<int> val(input.begin(), input.end());
+    return processor_.DecodeIds(val);
+  }
+
   int64_t GetPieceSize() const { return processor_.GetPieceSize(); }
+
+  int64_t PieceToId(const std::string &piece) const {
+    return processor_.PieceToId(piece);
+  }
+
+  std::string IdToPiece(const int64_t id) const {
+    return processor_.IdToPiece(id);
+  }
+
+  double GetScore(const int64_t id) const { return processor_.GetScore(id); }
+
+  bool IsUnknown(const int64_t id) const { return processor_.IsUnused(id); }
```

Comment:
Where does this come from?

Reply:
You can see the list of methods defined on the original SentencePieceProcessor class here. [sentencepiece.py](https://github.com/google/sentencepiece/blob/e8a84a16d13e8bf92892a1cd92e4de3b0d0321fd/python/sentencepiece.py#L101)

Reply:
Why do we need all the methods here? My understanding is that uses will usually load a sp model and apply it as a tokenizer.

Reply:
This is an attempt to keep BC change minimum. Before we introduce the C++ extension, all the methods were available.
Since we are adding the methods that are not used in library such as `__len__` and `__getitem__`, I thought it's a good idea to add other methods.

Reply:
Ah maybe you wanted to call ```processor_.IsUnknown(id)``` instead of ```processor_.IsUnused(id)``` for the definition of ```bool IsUnknown(const int64_t id)```?

Reply:
Good catch. will fix.

Reply:
fixed.